tests/istio.mk

```diff
@@ -83,35 +83,82 @@ e2e_all: | $(JUNIT_REPORT)
 	$(MAKE) e2e_simple e2e_mixer e2e_bookinfo \
 	|& tee >($(JUNIT_REPORT) > $(JUNIT_E2E_XML))
 
+# Run the e2e tests, with auth enabled. A separate target is used for non-auth.
 e2e_pilot: istioctl generate_yaml
 	go test -v -timeout 20m ./tests/e2e/tests/pilot ${TESTOPTS} -hub ${HUB} -tag ${TAG}
 
+e2e_pilot_noauth: istioctl generate_yaml
+	go test -v -timeout 20m ./tests/e2e/tests/pilot ${E2E_ARGS} -hub ${HUB} -tag ${TAG} --skip-cleanup -mixer=true -auth=disable -use-sidecar-injector=false
+
+test/minikube/auth/e2e_simple:
+	go test -v -timeout 20m ./tests/e2e/tests/simple -args --auth_enable=true \
+	  --skip_cleanup  -use_local_cluster -cluster_wide -test.v \
+	  ${E2E_ARGS} ${EXTRA_E2E_ARGS} \
+           ${TESTOPTS}
+           #|& tee >($(JUNIT_REPORT) > ${OUT_DIR}/tests/test-report-auth-simple.xml)
+
+test/minikube/noauth/e2e_simple:
+	go test -v -timeout 20m ./tests/e2e/tests/simple -args --auth_enable=false \
+	  --skip_cleanup  -use_local_cluster -cluster_wide -test.v \
+	  ${E2E_ARGS} ${EXTRA_E2E_ARGS} \
+           ${TESTOPTS}
+           #|& tee >($(JUNIT_REPORT) > ${OUT_DIR}/tests/test-report-noauth-simple.xml)
+
 # Target for running e2e pilot in a minikube env. Used by CI
-test/minikube/auth/e2e_pilot: istioctl generate_yaml
+test/minikube/auth/e2e_pilot: istioctl
 	mkdir -p ${OUT_DIR}/logs
-	kubectl create ns istio-system || true
-	kubectl create ns istio-test || true
+	mkdir -p ${OUT_DIR}/tests
+	kubectl create ns pilot-auth-system || true
+	kubectl create ns pilot-auth-test || true
 	go test -test.v -timeout 20m ./tests/e2e/tests/pilot -args \
 		-hub ${HUB} -tag ${TAG} \
 		--skip-cleanup --mixer=true --auth_enable=true \
 		-errorlogsdir=${OUT_DIR}/logs \
 		--use-sidecar-injector=false \
 		--core-files-dir=${OUT_DIR}/logs \
-         --ns istio-system \
-        -n istio-test \
+		--auth_enable=true \
+		--ns pilot-auth-system \
+		-n pilot-auth-test \
+		   ${TESTOPTS}
+        #   |& tee >($(JUNIT_REPORT) > ${OUT_DIR}/tests/test-report-auth-pilot.xml)
+
+
+# Target for running e2e pilot in a minikube env. Used by CI
+test/minikube/noauth/e2e_pilot: istioctl
+	mkdir -p ${OUT_DIR}/logs
+	mkdir -p ${OUT_DIR}/tests
+	# istio-system and pilot system are not compatible. Once we merge the setup it should work.
+	kubectl create ns pilot-system-test || true
+	kubectl create ns pilot-test || true
```

Comment:
we should refactor out this common test setup into its own target that these invocations can share (not blocking on this PR, just a longer term improvement we should make).

Reply:
Yes, creating a stable test configuration first - and running the test after without Sleep is quite good, but let's not do everything in one PR. Big +1 if you want to do it.